### PR TITLE
[FEATURE] On-ramp: Add buy-crypto deeplink

### PR DESCRIPTION
**Pull request by wachunei**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This PR adds support for `metamask://buy-crypto` and  `https://metamask.app.link/buy-crypto`, these will open the buy flow when the current network supports it.

**Screenshots/Recordings**

~~_If applicable, add screenshots and/or recordings to visualize the before and after of your change_~~

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by lorenzosantos** [comment]
@wachunei, can we configure the `https://metamask.app.link/buy-crypto` link to open the Pdapp `https://portfolio.metamask.io/buy/` if the user is clicking on the desktop?

File: app/core/DeeplinkManager.js [hunk 1]
```diff
@@ -186,6 +187,14 @@ class DeeplinkManager {
     });
   }
 
+  _handleBuyCrypto() {
+    if (allowedToBuy(this.network)) {
+      this.navigation.navigate(Routes.FIAT_ON_RAMP_AGGREGATOR.ID);
+    } else {
```
[the remark this answers is not in the document]
Comment (wachunei) [reply app/core/DeeplinkManager.js:193]: Fixed in a0adcc6c4

**Comment by Cal-L** [comment]
@wachunei Could you provide a video of this feature in action? Is it also intended to work for both tapping on a link as well as scanning a QR code?

**Comment by wachunei** [comment]
@Cal-L sadly no, the simulator was not handling any of the deeplinks correctly. 

> Is it also intended to work for both tapping on a link as well as scanning a QR code?

Yes, these two should work:
imageimage

**Comment by worldlyjohn** [comment]
@lorenzosantos might also make sense to trigger an event here with referrer to know if this is being used and where their coming from (assuming user opted-in)

**Review by cortisiko — DISMISSED** [review]
swinging this back your way @wachunei 


1
When I attempt to deep link to a contract approval: https://metamask.app.link/approve/0x514910771AF9Ca656af840dff83E8264EcF986CA@1/approve?address=0x178e3e6c9f547A00E33150F7104427ea02cfc747&uint256=3e18


 I get `TypeError: undefined is not can object (evaluating ‘’t.toLowerCase;)` 
See recording: http://recordit.co/WSXnJIi1ro 

2
When I deep link to the onramps flow via the URL SCHEMA deep link, then try to deep link to a dapp, The onramps flow always appear prior to opening the dapp.

See recording: 
https://recordit.co/dDdpvRSzUa

To reproduce:
Deep link here metamask://buy-crypto 
Tap get started
Then tap on “cancel”  
Now kill the app and attempt to deep link here: https://metamask.app.link/app.olympusdao.finance/#/bonds 
Notice the app opens to the onramps screen
If you were to tap cancel, then you are taken to the dapp in the browser

**Comment by wachunei** [comment]
1. Can you reproduce the error on main? test cases before 45s mark seem OK, for the last one I see no reason for this code to cause it 🤔 .

2. I see, this seems like the deeplink is not marked as handled, maybe I am missing a `handled()` call. @andreahaku I'd appreciate if you can review this code change please, I might be missing something 🙏 .

**Review by andreahaku — APPROVED** [review]
Can't reproduce the issue as well. Checked the code, tested the deep link and universal link, tried to reproduce the issue following the instruction by @cortisiko  and everything work correctly as expected. LGTM

**Comment by chrisleewilcox** [comment]
@wachunei I am seeing the Your Region screen constantly pop back up if user cancels out of it and never continues pass Your Region.
https://recordit.co/iy7Gfm09zu

**Comment by wachunei** [comment]
@chrisleewilcox is the browser loading the deeplink again and again? Can you kill the browser from the task manager? 

@andreahaku how can we solve the issue? If the above is not the case then the deeplink is not being marked as handled although `handled()` is being called.

**Comment by chrisleewilcox** [comment]
Issue #1 reported from @cortisiko is resolved.

Issue #2 reported from @cortisiko  is resolved.

Payment deeplink device browser tap
- iOS
- metamask://buy-crypto: PASS
- https://metamask.app.link/buy-crypto: PASS

- Android
- metamask://buy-crypto: PASS
- https://metamask.app.link/buy-crypto: PASS


Payment deeplink device in-app browser tap
- iOS
- metamask://buy-crypto: PASS
- https://metamask.app.link/buy-crypto: PASS

- Android
- metamask://buy-crypto: PASS
- https://metamask.app.link/buy-crypto: PASS


Regression:
Send Flow deeplink: PASS
URL deeplink: PASS
QR Scan URL deeplink: PASS

**Review by chrisleewilcox — APPROVED** [review]
LGTM

**Review by chrisleewilcox — APPROVED** [review]

**Pull request merged**